scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3252,11 +3260,16 @@ void SlaOutputDev::updateFont(GfxState *state)
 		fontsrc->unref();
 }
 
-void SlaOutputDev::drawChar(GfxState *state, double x, double y, double dx, double dy, double originX, double originY, CharCode code, int nBytes, POPPLER_CONST_082 Unicode *u, int uLen)
+void SlaOutputDev::drawChar(GfxState* state, double x, double y, double dx, double dy, double originX, double originY, CharCode code, int nBytes, POPPLER_CONST_082 Unicode* u, int uLen)

```

Comment:
i think that you need a `drawCharAsVector()` and a `drawCharAsText()` function. the two branches in this function have (almost) nothing in common.

Reply:
I can seperate out the existing coder in drawChar so that the vector code is in it's own function which gets called. I tried not to make too many changes to the existing code.